readinglist/tests/test_utils.py

```diff
@@ -36,3 +36,13 @@ def test_does_remove_middle_spaces(self):
 
     def test_idempotent_for_null_values(self):
         self.assertEqual(strip_whitespace(colander.null), colander.null)
+
+
+class ResourceIdTest(unittest.TestCase):
+    def test_string_resource_id(self):
+        self.assertEqual(resource_id('toto'), 'toto')
```

Comment:
I thought that the resource id might be a property/method of the resource class instead.

For example, if we imagine "dynamic resource of cloud storage" : 

```python
@crud(collection_path='/{resource_id}',
      item_path='/{resource_id}/{id}')
class DynamicResource(...):
 ....
    def __init__(...):
        self.resource_id = self.request.match_dict['resource_id']
```